app/src/main/java/org/mifos/mobile/viewModels/RegistrationViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package org.mifos.mobile.viewModels
+
+import android.util.Patterns
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.schedulers.Schedulers
+import okhttp3.ResponseBody
+import org.mifos.mobile.api.DataManager
+import org.mifos.mobile.models.register.RegisterPayload
+import javax.inject.Inject
+
+class RegistrationViewModel @Inject constructor(var dataManager : DataManager?) : ViewModel() {
+    private val compositeDisposables: CompositeDisposable = CompositeDisposable()
+
+    private var registrationResultSuccess = MutableLiveData<Boolean>()
+    val readRegistrationResultSuccess : LiveData<Boolean> get() = registrationResultSuccess
+    private var exceptionOnRegistration : Throwable? = null
+    val readExceptionOnRegistration : Throwable get() = exceptionOnRegistration!!
+
+    fun isInputFieldBlank(fieldText : String) : Boolean {
+        return fieldText.trim { it <= ' '}.isEmpty()
+    }
+
+    fun isInputLengthInadequate(fieldText : String) : Boolean {
+        return fieldText.trim { it <= ' '}.length < 6
```

Comment:
Is this in Kolinish way? :)

Reply:
No it isn't, it was legacy code hence used it the same way. I'll replace _fieldText.trim { it <= ' ' }_ with **fieldText.trim()**